consensus/merge/src/main/java/org/hyperledger/besu/consensus/merge/blockcreation/MergeCoordinator.java

```diff
@@ -398,27 +398,37 @@ public void awaitCurrentBuildCompletion(final PayloadIdentifier payloadId, final
       return;
     }
 
-    final CompletableFuture<Void> future = task.getBlockCreationFuture();
-    if (future == null) {
-      LOG.debug("No future found for payload {}", payloadId);
+    final CompletableFuture<Void> blockCreationFuture = task.getBlockCreationFuture();
+    if (blockCreationFuture == null) {
+      LOG.debug("No blockCreationFuture found for payload {}", payloadId);
       return;
     }
 
+    final long startTime = System.currentTimeMillis();
```

Comment:
just a comment for sharing what I learnt from @ahamlat, that explained to me how using nanos to calculate elapsed time is more reliable that `System.currentTimeMillis()`